code/Pagetypes/FreshdeskPage.php

```diff
@@ -0,0 +1,238 @@
+            $productID = FRESHDESK_PRODUCT_ID;
+        }
+
+        // Get all open by default
+        $filter = ['status' => 2];
+        if ($this->request->getVar('status')) {
+            $filter['status'] = $this->request->getVar('status');
+        }
+        if ($this->request->getVar('priority')) {
+            $filter['priority'] = $this->request->getVar('priority');
+        }
+        $filter = $this->validateFilter($filter);
+
+        $tickets = $this->freshdeskTicketFilter($tickets, $productID, $filter);
+        $tickets = $this->humanReadable($tickets);
+        $tickets = new ArrayList($tickets);
+        return new PaginatedList($tickets, $this->request);
+    }
+
+    /**
+    * Filters tickets for a user based on product ID or other filter
+    *
+    * @param Array $tickets, String $productID, Array $filter
+    * @return Array $tickets
+    */
+    private function freshdeskTicketFilter($tickets, $productID = false, $filter)
+    {
+        // fast return if nothing to filter on
+        if (!$productID && !$filter) {
+            return $tickets;
+        }
+
+        foreach ($tickets as $key => $val) {
+            if ($productID && $val['product_id'] != $productID) {
+                unset($tickets[$key]);
+                continue;
+            }
+
+            if (empty($filter) || count($filter) == 0) {
+                continue;
+            }
+
+            foreach ($filter as $filterKey => $filterVal) {
+                $doFilter = $this->doFilter($val, $filterKey, $filterVal);
+                if ($doFilter) {
+                    unset($tickets[$key]);
+                }
+            }
+        }
+        return $tickets;
+    }
+
+    /**
+    * Return true or false, depending on if a ticket should be removed based on a match to a filter
+    *
+    * @param Array $ticket, String $filterKey, String $filterVal
+    * @return Boolean
+    */
+    private function doFilter($ticket, $filterKey, $filterVal)
+    {
+        if ($ticket[$filterKey] == (int) $filterVal) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+    * Converts the numeric values in a ticket to something that makes sense to a human. Mappings defined as statics on this class
+    *
+    * @param Array $tickets
+    * @return Array $formattedTickets
+    */
+    private function humanReadable($tickets)
+    {
+        $priorities = Config::inst()->get('FreshdeskPage_Controller', 'freshdeskPriority');
+        $statuses =  FreshdeskTicketStatuses::get();
+        $agents = FreshdeskAgents::get();
+
+        $formattedTickets = [];
+        foreach ($tickets as $ticket) {
+            if (isset($agents->find('AgentId', $ticket['responder_id'])->Name)) {
+                $responder = $agents->find('AgentId', $ticket['responder_id'])->Name;
+            } else {
+                $responder = '';
+            }
+            $status = $statuses->find('StatusId', $ticket['status'])->Name;
```

Comment:
No isset on this one? can you add some checking here like you had with agent? @McJoppy